### PR TITLE
25 select date

### PR DESCRIPTION
**Pull request by Fredmpk**
It now  uses date-fns to get all dates of the next 7 days, querys all times that match on the dates and filters the dates that have all seats booked as in your previous PR and returns an object with the dates as key and the times as values.

### CODE REVIEW

File: routes/times.ts [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
+  const unformattedDatesNextWeek = eachDayOfInterval({
+    start: new Date(today),
+    end: new Date(addDays(today, 6)),
+  });
+
+  const datesNextWeek = unformattedDatesNextWeek.map((date) =>
+    format(date, "dd-MM-yyyy")
+  );
+
+  const screenings = await sql<
```
Comment (dnmct) [inline routes/times.ts]: Please use prisma

**Review by dnmct — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma [hunk 2]
```diff
@@ -14,19 +14,27 @@ datasource db {
...
   name String
 
   owners User[]
 }
+
+model Screening {
+  id           String @id @default(uuid())
+  date         String
+  time         String
+  booked_seats String
```
Comment (louziferr) [inline prisma/schema.prisma]: Please use Camel Casing for field names.

File: prisma/seed.ts [hunk 3]
```diff
@@ -33,12 +33,77 @@ async function main() {
     { name: "Elder Scrolls Online" },
   ];
 
+  const screenings: Prisma.ScreeningCreateInput[] = [
+    {
+      id: "e72f636c-7bcb-4c35-8b77-c1c066f9b0a1",
```
Comment (louziferr) [inline prisma/seed.ts]: UUIDs are created by prisma, and should not be hard-coded.

**Review by louziferr — CHANGES_REQUESTED** [review]
small changes needed

File: prisma/schema.prisma [hunk 4]
```diff
@@ -14,19 +14,27 @@ datasource db {
...
   name String
 
   owners User[]
 }
+
+model Screening {
+  id          String @id @default(uuid())
+  date        String
+  time        String
+  bookedSeats String
```
Comment (louziferr) [inline prisma/schema.prisma]: bookedSeats should be a string array

File: prisma/seed.ts [hunk 5]
```diff
@@ -33,12 +33,69 @@ async function main() {
     { name: "Elder Scrolls Online" },
   ];
 
+  const screenings: Prisma.ScreeningCreateInput[] = [
+    {
+      date: "24-07-2024",
+      time: "10:00",
+      bookedSeats: "A1,B1,C1,D1,E1,F1",
```
Comment (louziferr) [inline prisma/seed.ts]: bookedSeats should be a string array here as well

File: routes/times.ts [hunk 6]
```diff
@@ -1,29 +1,50 @@
...
+    const availableTimes = screenings.reduce<{ [key: string]: string[] }>(
+      (acc, screening) => {
+        if (!acc[screening.date]) {
+          acc[screening.date] = [];
+        }
 
-  const availableTimes = db.screening
-    .filter((screening: Screening) => screening.bookedSeats.length < SeatsTotal)
-    .map((screening: Screening) => screening.time);
+        if (screening.booked_seats.split(",").length < SeatsTotal) {
```
Comment (louziferr) [inline routes/times.ts]: this causes an error. please fix :)

**Review by dnmct — APPROVED** [review]

**Review by dnmct — APPROVED** [review]

**Pull request merged**